backend/src/main/java/gov/cdc/usds/simplereport/service/TestResultUploadService.java

```diff
@@ -112,4 +149,27 @@ private UploadStatus parseStatus(ReportStreamStatus status) {
         return UploadStatus.FAILURE;
     }
   }
+
+  public UploadResponse getUploadSubmission(UUID id)
+      throws InvalidBulkTestResultUploadException, InvalidRSAPrivateKeyException {
+    Organization org = _orgService.getCurrentOrganization();
+
+    TestResultUpload result =
+        _repo
+            .findByInternalIdAndOrganization(id, org)
+            .orElseThrow(InvalidBulkTestResultUploadException::new);
+
+    String reportingScope = createReportingScope(simpleReportCsvUploadClientName);
+
+    Map<String, String> queryParams = new LinkedHashMap<>();
+    queryParams.put("scope", reportingScope);
+    queryParams.put("grant_type", "client_credentials");
+    queryParams.put(
+        "client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer");
+    queryParams.put("client_assertion", createDataHubSenderToken(signingKey));
+
+    TokenResponse r = _client.fetchAccessToken(queryParams);
```

Comment:
are we requesting a fresh token on every request? seems like we should be caching it

Reply:
I think that's a good call. The volume of these requests will be pretty low in the near-term so maybe we can call that a nice-to-have for the initial release. I'll make a ticket